# Assignment 1: Python Fundamentals and Dataset Exploration
*Course:* GRTR 513 Decoding Python to Fuel Your Research  
*Due:* **June 26, 2026, 1:00 pm**  
*Estimated time:* ≈ 1.5-2 hours

> **Instructions:** This assignment focuses on **Python fundamentals**: variables, data types, comparison operators, boolean operators, if statements, for loops, and file operations.  
> **You will work with a dataset of your choice** (see the course Moodle for a selection of datasets), load it, inspect it, and perform basic operations.
> **Important:** If you choose to work with a dataset that's not available in the Moodle, you **must** upload it together with your noetbook submission.

**Fill this in!** *(You can double-click any notebook cell to edit)*

---

**Student Name:** Tanzir Bin Razzaque 
**Chosen Dataset:** exoplanets  
**Dataset Source (URL or filename):** [exoplanets.csv](https://moodle.concordia.ca/moodle/pluginfile.php/7929315/mod_folder/content/0/exoplanets.csv?forcedownload=1)

---

## Learning Objectives
- Use **variables** to store and manipulate data
- Work with different **data types** (strings, integers, floats, booleans)
- Use **comparison operators** and **boolean operators** to make decisions
- Write **if statements** to control program flow
- Use **for loops** with the `range()` function to iterate
- Perform **file operations** (reading and writing)
- Load and inspect a real dataset

### What's Allowed / Not Allowed
- **Allowed**: All basic Python covered in Weeks 1-2
- **Not allowed**: Pandas, NumPy, Matplotlib, or other external libraries

## Part 1: Variables and Data Types (20 points)

### Task 1.1: Dataset Information Variables (10 points)
Create variables to store information about your chosen dataset. Include:
- Dataset name (string)
- Number of expected rows (integer - you can estimate)
- Number of expected columns (integer)
- Dataset source/URL (string)
- Whether you've downloaded it (boolean)

Print all variables with descriptive labels (remember you can use f-strings for this!).

For this part, you don't have to open the dataset with Python. You can inspect it in other software like Excel or Google Sheets and then save the number of rows and columns you saw or estimated as variables in Python.

In [22]:
# Task 1.1: YOUR CODE HERE
dataset_name = "exoplanets"
rows = 49
columns = 1964
dataset_source = "https://moodle.concordia.ca/moodle/pluginfile.php/7929315/mod_folder/content/0/exoplanets.csv?forcedownload=1"
downloaded_dataset = True

print(f"Name of the dataset is {dataset_name}")
print(f"Number of rows in the dataset: {rows}")
print(f"Numer of columns in the dataset: {columns}")
print(f"Dataset URL: {dataset_source}")
print(f"Downloaded Dataset: {downloaded_dataset}")





Name of the dataset is exoplanets
Number of rows in the dataset: 49
Numer of columns in the dataset: 1964
Dataset URL: https://moodle.concordia.ca/moodle/pluginfile.php/7929315/mod_folder/content/0/exoplanets.csv?forcedownload=1
Downloaded Dataset: True


Once you're done, **double-click this Markdown cell** and edit the two bullet points below to give some context about your dataset.

* The dataset I chose is exoplanets
* I chose this dataset because I want to know more about the properties of the planets that are outside of our solarsystem

### Task 1.2: Working with Data Types (10 points)
Using your dataset information variables:
1. Print the **type** of each variable using the `type()` function
2. Create a new variable that combines the dataset name and source using string concatenation
3. Calculate the total number of data points (rows × columns) and store it in a variable

For point 3, you don't have to open the data with Python for this. You can inspect it in other software like Excel or Google Sheets and then do the rows × columns multiplication in Python.

In [25]:
# Task 1.2: YOUR CODE HERE
print(f"Dataset name type: {type(dataset_name)}")
print(f"Dataset row type: {type(rows)}")
print(f"Dataset column type: {type(columns)}")
print(f"Dataset Source type: {type(dataset_source)}")
print(f"downloaded_dataset type: {type(downloaded_dataset)}")

print(f"Dataset Name: {dataset_name}, source: {dataset_source}")

total_data_points = rows * columns
print(f"Total data points: {total_data_points}")







Dataset name type: <class 'str'>
Dataset row type: <class 'int'>
Dataset column type: <class 'int'>
Dataset Source type: <class 'str'>
downloaded_dataset type: <class 'bool'>
Dataset Name: exoplanets, source: https://moodle.concordia.ca/moodle/pluginfile.php/7929315/mod_folder/content/0/exoplanets.csv?forcedownload=1
Total data points: 96236


## Part 2: Comparison and Boolean Operators (20 points)

### Task 2.1: Dataset Size Comparisons (10 points)
Write code that:
1. Checks if the number of rows is greater than 100
2. Checks if the number of columns is less than or equal to 10
3. Checks if the total data points is not equal to 1000

Print the result of each comparison with a descriptive message. You may use the variables you created in Part 1 for this!

In [14]:
# Task 2.1: YOUR CODE HERE
#Checks if num of rows > 100
print(f"Number of rows > 100: {rows > 100}")
#Checks if num of cols <= 10
print(f"Number of columns <= 10: {columns <= 10}")
#Checks total data points != 1000
print(f"Total data points != 1000: {total_data_points != 1000}")


Number of rows > 100: False
Number of columns <= 10: False
Total data points != 1000: True


### Task 2.2: Boolean Operators (10 points)
Using boolean operators (`and`, `or`, `not`):
1. Check if the dataset has more than 50 rows **AND** less than 20 columns
2. Check if the dataset has more than 1000 rows **OR** more than 50 columns
3. Check if you have **NOT** downloaded the dataset yet

Print the result of each check.

You may use the variables that you've created above here!

In [15]:
# Task 2.2: YOUR CODE HERE
print(f"Dataset has more than 50 rows and less than 20 columns: {rows > 50 and columns < 20}")
print(f"Dataset has more than 1000 rows or less than 50 colums: {rows > 1000 or columns > 50}")
print(f"Dataset downloaded: {downloaded_dataset}")






Dataset has more than 50 rows and less than 20 columns: False
Dataset has more than 1000 rows or less than 50 colums: True
Dataset downloaded: True


## Part 3: If Statements (20 points)

### Task 3.1: Dataset Size Classification (20 points)
Write a series of if statements to classify your dataset size:
- If total data points > 10,000: print "This is a large dataset"
- If total data points is between 1,000 and 10,000: print "This is a medium dataset"
- If total data points < 1,000: print "This is a small dataset"

Also check:
- If the dataset has been downloaded, print "Ready to analyze!"
- If not downloaded, print "Please download the dataset first"

You may use the variables you've created above for this!

In [16]:
# Task 3.1: YOUR CODE HERE

if total_data_points > 10000:
    print("This is a large dataset")
elif total_data_points >= 1000 and total_data_points <= 10000:
    print("This is a medium dataset")
else:
    print("This is a small dataset")

if downloaded_dataset:
    print("Ready to analyze!")
else:
    print("Please download the dataset first")




This is a large dataset
Ready to analyze!


## Part 4: For Loops with Range (20 points)

### Task 4.1: Row Counter (10 points)
Use a `for` loop with the `range()` function to:
- Print "Processing row X" for the first 10 rows of your dataset
- X should be the row number (1, 2, 3, ... 10)

You don't have to do any actual processing here! The purpose of this task is for you to get familiar with the `for` loop syntax and usage.

In [17]:
# Task 4.1: YOUR CODE HERE
print("Processing row X: ")
for i in range (1,11):
    print(i)






Processing row X: 
1
2
3
4
5
6
7
8
9
10


### Task 4.2: Sum Calculation (10 points)
Use a `for` loop to calculate the sum of all numbers from 1 to the number of columns in your dataset.

Print the result.

In [18]:
# Task 4.2: YOUR CODE HERE
sum = 0
for i in range (1, columns + 1):
    sum += i
print(sum)







1929630


## Part 5: File Operations and Dataset Loading (20 points)

### Task 5.1: Create Dataset Info File (10 points)
Create a text file called `dataset_info.txt` that contains:
- Dataset name
- Number of rows
- Number of columns
- Data source
- Date you started working with it (you can use today's date as a string)

Use the `with open()` statement with `mode='w'` to write the file.

In [19]:
# Task 5.1: YOUR CODE HERE
with open("dataset_info.txt", mode="w") as file:
    file.write(
        f"Dataset name: {dataset_name}\n"
        f"Number of rows: {rows}\n"
        f"Number of columns: {columns}\n"
        f"Data source: {dataset_source}\n"
        f"Date: 25 June 2026"
    )






### Task 5.2: Read and Print the File (10 points)
Read the `dataset_info.txt` file you just created and print its contents.

Use the `with open()` statement with `mode='r'`.

In [20]:
# Task 5.2: YOUR CODE HERE
with open("dataset_info.txt", mode = "r") as file:
    contents = file.read()
print(contents)





Dataset name: exoplanets
Number of rows: 49
Number of columns: 1964
Data source: https://moodle.concordia.ca/moodle/pluginfile.php/7929315/mod_folder/content/0/exoplanets.csv?forcedownload=1
Date: 25 June 2026


## Part 6: Load and Inspect Your Dataset (Bonus: 10 points)

### Task 6.1: Load Your Dataset
Load your actual dataset file and print out its contents. You can:
- Use `with open()` with `mode='r'`
- Read the file using `.read()` or `.readlines()`
- Print the first 10-20 lines to inspect the data

**Note:** Just print it out - we're not doing any fancy processing yet!

In [36]:
# Task 6.1: YOUR CODE HERE
# Replace 'your_dataset.csv' with your actual filename
with open("./data/exoplanets.csv", mode = "r") as f:
    for i in range(10):
        contents = f.readline()
        if not contents:
            break
        print(contents.strip())







kepid,kepoi_name,kepler_name,koi_disposition,koi_pdisposition,koi_score,koi_fpflag_nt,koi_fpflag_ss,koi_fpflag_co,koi_fpflag_ec,koi_period,koi_period_err1,koi_period_err2,koi_time0bk,koi_time0bk_err1,koi_time0bk_err2,koi_impact,koi_impact_err1,koi_impact_err2,koi_duration,koi_duration_err1,koi_duration_err2,koi_depth,koi_depth_err1,koi_depth_err2,koi_prad,koi_prad_err1,koi_prad_err2,koi_teq,koi_teq_err1,koi_teq_err2,koi_insol,koi_insol_err1,koi_insol_err2,koi_model_snr,koi_tce_plnt_num,koi_tce_delivname,koi_steff,koi_steff_err1,koi_steff_err2,koi_slogg,koi_slogg_err1,koi_slogg_err2,koi_srad,koi_srad_err1,koi_srad_err2,ra,dec,koi_kepmag
10797460,K00752.01,Kepler-227 b,CONFIRMED,CANDIDATE,1,0,0,0,0,9.48803557,2.78E-05,-2.78E-05,170.53875,2.16E-03,-2.16E-03,0.146,0.318,-0.146,2.9575,0.0819,-0.0819,6.16E+02,1.95E+01,-1.95E+01,2.26,2.60E-01,-1.50E-01,793,,,93.59,29.45,-16.65,35.8,1,q1_q17_dr25_tce,5455,81,-81,4.467,0.064,-0.096,0.927,0.105,-0.061,291.93423,48.141651,15.347
10797460,K00752.0

---

## Submission Checklist
- [ ] **Part 1:** Variables and data types completed
- [ ] **Part 2:** Comparison and boolean operators completed
- [ ] **Part 3:** If statements completed
- [ ] **Part 4:** For loops with range completed
- [ ] **Part 5:** File operations completed (created and read file)
- [ ] **Part 6 (Bonus):** Dataset loaded and inspected
- [ ] All code cells run without errors
- [ ] Student name and dataset information filled in at the top
- [ ] Notebook (.ipynb file) and dataset uploaded to Moodle

### Grading Breakdown (100 points + 10 bonus)
- **Part 1: Variables and Data Types (20 points)**
  - Task 1.1: Dataset information variables (10 points)
  - Task 1.2: Working with data types (10 points)

- **Part 2: Comparison and Boolean Operators (20 points)**
  - Task 2.1: Dataset size comparisons (10 points)
  - Task 2.2: Boolean operators (10 points)

- **Part 3: If Statements (20 points)**
  - Task 3.1: Dataset size classification (20 points)

- **Part 4: For Loops with Range (20 points)**
  - Task 4.1: Row counter (10 points)
  - Task 4.2: Sum calculation (10 points)

- **Part 5: File Operations (20 points)**
  - Task 5.1: Create dataset info file (10 points)
  - Task 5.2: Read and print the file (10 points)

- **Part 6: Load and Inspect Dataset (Bonus: 10 points)**
  - Task 6.1: Load dataset (5 points)
  - Task 6.2: Write observations (5 points)

### Tips for Success
- Test each code cell as you go
- Use descriptive variable names
- Add comments to explain what your code does
- Don't forget to save your work frequently!

### Academic Integrity
Work individually. You may discuss general approaches with classmates, but all code must be your own.